scm/driver/bitbucket/git.go

```diff
@@ -91,9 +104,7 @@ func (s *gitService) CompareChanges(ctx context.Context, repo, source, target st
 type branch struct {
 	Type   string `json:"type"`
 	Name   string `json:"name"`
-	Target struct {
-		Hash string `json:"hash"`
-	} `json:"target"`
+	Target *commit
```

Comment:
do we need to change this?  in general, we try to avoid pointers in this library.  If we are going to introduce a pointer, we need to find everywhere the pointer is referenced and check for a null value. So if we don't need to change, I'd recommend leaving inline.

Reply:
This is needed for convertCommit where it requires it the be the commit. I can write another convertCommit if it is better?